contracts/citycoin-core.clar

```diff
@@ -1,201 +1,539 @@
-  principal          ;; address
+;; returns set city wallet principal
+(define-read-only (get-city-wallet)
+  (var-get cityWallet)
 )
+ 
+;; protected function to update city wallet variable
+(define-public (set-city-wallet (newCityWallet principal))
+  (begin
+    (asserts! (is-authorized-city) (err ERR_UNAUTHORIZED))
+    (ok (var-set cityWallet newCityWallet))
+  )
+)
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; CONTRACT MANAGEMENT
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;; contract states used in map CityCoinContracts
+(define-constant CONTRACT_DEFINED u0)
+(define-constant CONTRACT_ACTIVE u1)
+(define-constant CONTRACT_INACTIVE u2)
+
+;; initial value for active logic contract
+;; set to deployer address at startup to prevent circular dependency of citycoin-core on citycoin-logic
+(define-data-var activeContract principal CONTRACT_OWNER)
+
+;; defines if the initial logic contract is provided for initialization
+(define-data-var initialized bool false)
 
-;; used as Proposal ID
-(define-data-var proposalNonce uint u0)
+;; updates the activeContract variable with initial logic contract
+;; can only be run once after deployment
+(define-public (setup-active-contract (firstActiveContract principal))
+  (begin
+    (asserts! (is-authorized-owner) (err ERR_UNAUTHORIZED))
+    (asserts! (not (var-get initialized)) (err ERR_UNAUTHORIZED))
+    (var-set activeContract firstActiveContract)
+    (var-set initialized true)
+    (ok true)
+  )
+)
 
-(define-map Contracts
-  principal  ;; MiningContract address
-  { proposalId: uint, name: (string-ascii 100) }
+;; returns active contract
+(define-read-only (get-active-contract)
+  (var-get activeContract)
 )
 
-(define-map Proposals
-  uint ;; Proposal id
+;; store logic contract information
+(define-map CityCoinContracts
+  principal
   {
-    address: principal,
-    startBH: uint,
-    endBH: uint,
-    voters: uint,
-    votes: uint,
-    isOpen: bool
+    state: uint,
+    startHeight: uint,
+    endHeight: uint,
+    active: bool
   }
 )
 
-;; check
-(define-map ProposalVoters
-  { proposalId: uint, voter: principal }
-  bool
+;; returns logic contract information
+(define-read-only (get-contract (address principal))
+  (map-get? CityCoinContracts address)
 )
 
-(define-read-only (get-proposals-count)
-  (var-get proposalNonce)
+;; set initial value for logic contract
+(map-set CityCoinContracts
+  .citycoin-logic-v1
+  {
+    state: CONTRACT_DEFINED,
+    startHeight: u0,
+    endHeight: u0,
+    active: false 
+  })
+
+;; TODO: function to update active contract
+;;   called by register miner to activate first contract
+;;   called by city wallet to update to new contract (+ delay)
+;; (try! (contract-call? .citycoin-logic-v1 startup (some activationBlockVal)))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; REGISTRATION
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(define-data-var activationBlock uint u340282366920938463463374607431768211455)
```

Comment:
For what's in there so far, yes I agree, but there are checks added to the mining claim function that ensure the current block height is higher than this value - which is part of why we maxed it out. Will definitely keep this in mind during the refactor though and there may be a better way to perform that check so this could be `u0`.  I agree it's a little messy.